### PR TITLE
feat: traffic visibility UI and store

### PR DESCRIPTION
**Pull request by daveleek**
## About the changes

This PR provides store method for retrieving traffic usage data based on period parameter, and UI + ui hook with the new chart for displaying traffic usage data spread out over selectable month.

Skjermbilde 2024-03-21 kl  12 40 38

In this PR we copied and adapted a plugin written by DX for highlighting a column in the chart:

image

There are some minor improvements planned which will come in a separate PR, reversing the order in legend and tooltip so the colors go from light to dark, and adding a month -sum below the legend

## Discussion points

- Should any of this be extracted as a separate reusable component?

### CODE REVIEW

File: frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+    alignItems: 'center',
+    fontSize: theme.fontSizes.bodySize,
+    margin: 0,
+    marginTop: theme.spacing(1),
+    fontWeight: theme.fontWeight.bold,
+}));
+
+const padMonth = (month: number): string => {
+    return month < 10 ? `0${month}` : `${month}`;
+};
```
Comment (nunogois) [inline frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx:58]: Can be simplified:

const padMonth = (month: number): string => month < 10 ? `0${month}` : `${month}`;

File: frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+        label:
+            label ||
+            date.toLocaleString('en-US', { month: 'long', year: 'numeric' }),
+    };
+};
+
+const getSelectablePeriods = (): SelectablePeriod[] => {
+    const current = new Date(Date.now());
+    const selectablePeriods = [toSelectablePeriod(current, 'Current month')];
+    for (let monthAdd = 1; monthAdd < 13; monthAdd++) {
```
Comment (nunogois) [inline frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx:79]: Maybe rename this variable to something like `monthRange`?

File: frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    endpointsInfo: Record<string, EndpointInfo>,
+): ChartDatasetType[] => {
+    if (!traffic || !traffic.usage || !traffic.usage.apiData) {
+        return [];
+    }
+
+    const data = traffic.usage.apiData
+        .filter((item: { apiPath: string }) => !!endpointsInfo[item.apiPath])
+        .sort(
+            (item1: any, item2: any) =>
```
Comment (nunogois) [inline frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx:119]: Can we get better types here?

File: frontend/src/hooks/api/getters/useInstanceTrafficMetrics/useInstanceTrafficMetrics.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import useSWR from 'swr';
+import { useMemo } from 'react';
+import { formatApiPath } from 'utils/formatPath';
+import handleErrorResponses from '../httpErrorResponseHandler';
+
+export interface IInstanceTrafficMetricsResponse {
+    usage: any;
```
Comment (nunogois) [inline frontend/src/hooks/api/getters/useInstanceTrafficMetrics/useInstanceTrafficMetrics.ts:7]: Can we get better types here?

**Review by nunogois — APPROVED** [review]
LGTM!

File: frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+                color: theme.palette.text.primary,
+                pointStyle: 'circle',
+                usePointStyle: true,
+                boxHeight: 6,
+                padding: 15,
+                boxPadding: 5,
+            },
+        },
+        tooltip: {
+            backgroundColor: '#fff',
```
[the remark this answers is not in the document]
Comment (daveleek) [reply frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx:198]: Very good point! I think we can!

File: frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx (hunk 3 above)
Comment (daveleek) [reply frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx:119]: I'll have a look!

File: frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx (hunk 2 above)
Comment (daveleek) [reply frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx:79]: Yeah I'll try to come up with something!

File: frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx (hunk 1 above)
Comment (daveleek) [reply frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx:58]: Good spot!

**Pull request merged**